packages/mask/background/services/site-adaptors/connect.ts

```diff
@@ -41,15 +42,27 @@ export async function connectSite(
 
     if (!(await requestSiteAdaptorsPermission([worker]))) return
 
+    // #region reset the global setup status setting
     currentSetupGuideStatus[network].value = stringify({
         status: type === 'nextID' ? SetupGuideStep.VerifyOnNextID : SetupGuideStep.FindUsername,
         persona: identifier.toText(),
         username: profile?.userId,
     })
 
     await delay(100)
+    // #endregion
 
-    if (worker.homepage) {
-        browser.tabs.create({ active: true, url: worker.homepage })
+    // #region open or switch the site
+    const openedTabs = await browser.tabs.query({ url: `${worker.homepage}/*` })
+    const targetTab = openedTabs.find((x) => x.active) ?? first(openedTabs)
+
+    if (targetTab?.id) {
+        await browser.tabs.update(targetTab.id, {
+            active: true,
+        })
+        await browser.windows.update(targetTab.windowId, { focused: true })
+    } else {
+        await browser.tabs.create({ active: true, url: worker.homepage })
     }
```

Comment:
Add a shared method `openOrActiveWindow()` in `packages/shared-base-ui/src/bom`.